### PR TITLE
feat(graphql): add aggregate information for active recordings in the GraphQL API

### LINKED ISSUE #1030: [Task] Add aggregate information for active recordings in the GraphQL API
In #924 the [GraphQL schema](https://github.com/cryostatio/cryostat/blob/22380ca0bd76689b3de162fb94487f1124d3775c/src/main/resources/types.graphqls#L86) for archived recordings was changed to include aggregate information alongside the actual recording list itself. 

```
type Recordings {
    active(filter: ActiveRecordingFilterInput): [ActiveRecording!]!
    archived(filter: ArchivedRecordingFilterInput): Archived!
}

...

type Archived {
    data: [ArchivedRecording!]!
    aggregate: AggregateInfo!
}

type AggregateInfo {
    count: Long!
}
```
Active recordings should mirror this change for consistency purposes. Adding more aggregate information fields alongside the count for both types of recordings would be nice as well.

### PR DESCRIPTION
**Pull request by maxcao13**
Fixes #1030

### CODE REVIEW

**Comment by andrewazores** [comment]
I think this will depend on a -web PR to be created as well: https://github.com/cryostatio/cryostat-web/pull/473#issuecomment-1223383346

**Comment by andrewazores** [comment]
Related: https://github.com/cryostatio/cryostat/pull/1077

@maxcao13 can you resume work on this PR so we can be sure it gets into this release alongside the linked one above? Otherwise the API structure becomes asymmetric between active and archived.

**Comment by maxcao13** [comment]
The PR doesn't included the new 'size' aggregate info, do you want me to include it here?

**Comment by maxcao13** [comment]
Also, a query like this:
```bash
$ http :8181/api/v2.2/graphql Authorization:"Basic $(echo user:pass | base64)" query="query { targetNodes { name recordings { active { aggregate { count } } }
 } }"
```
results in 
```
{
    "data": null,
    "errors": [
        {
            "extensions": {
                "classification": "DataFetchingException"
            },
            "locations": [
                {
                    "column": 28,
                    "line": 1
                }
            ],
            "message": "Exception while fetching data (/targetNodes[4]/recordings) : Failed to retrieve RMIServer stub: javax.naming.CommunicationException [Root exception is java.rmi.ConnectIOException: non-JRMP server at remote endpoint]",
            "path": [
                "targetNodes",
                4,
                "recordings"
            ]
        }
    ]
}
```
if a target is unreachable (in my case 9095 with no SSL cert), but this is fixed in the archive endpoint PR #1047, by the try catch in the `ActiveRecordingFetcher`.

This is a query on reachable targets using smoketest (for me who doesn't know how to validate the SSL for 9095).
```bash
$ http :8181/api/v2.2/graphql Authorization:"Basic $(echo user:pass | base64)" query="query { environmentNodes(filter: { name: \"quarkus-test\" }) { name nodeType descendantTargets { recordings { active { data { name state metadata { labels } } } archived { data { name metadata { labels } } } } } } }"
```

**Comment by andrewazores** [comment]
> The PR doesn't included the new 'size' aggregate info, do you want me to include it here?

I'm not sure if we can easily do that right now - the [JDK Recording class](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jfr/jdk/jfr/Recording.html#getSize()) has an accessor for this, but the [JMC IRecordingDescriptor](https://github.com/cryostatio/cryostat-core/blob/main/src/main/java/org/openjdk/jmc/rjmx/services/jfr/IRecordingDescriptor.java) does not. The information [is available over JMX](https://docs.oracle.com/en/java/javase/12/docs/api/jdk.management.jfr/jdk/management/jfr/FlightRecorderMXBean.html) with [an accessor like the Recording one](https://docs.oracle.com/en/java/javase/12/docs/api/jdk.management.jfr/jdk/management/jfr/RecordingInfo.html#getSize()), but to use this we would need to update `-core`. This could be something to propose for JMC to add upstream, too.

In theory this would be a nice addition but in practice I don't think we should go for it, not right now.

File: src/main/java/io/cryostat/net/web/http/api/v2/graph/ActiveRecordingsFetcher.java [hunk 1]
```diff
@@ -94,54 +99,69 @@ public List<GraphRecordingDescriptor> getAuthenticated(DataFetchingEnvironment e
...
+        return active;
+    }
+
+    static class Active {
+        List<GraphRecordingDescriptor> data;
+        AggregateInfo aggregate;
+    }
+
+    static class AggregateInfo {
+        Long count;
```
Comment (andrewazores) [inline src/main/java/io/cryostat/net/web/http/api/v2/graph/ActiveRecordingsFetcher.java:165]: Can this be the primitive `long`?

File: src/test/java/itest/GraphQLIT.java [hunk 2]
```diff
@@ -460,8 +461,11 @@ void testDeleteMutation() throws Exception {
         query.put(
                 "query",
                 "query { targetNodes(filter: { annotations: \"PORT == 9093\" }) { recordings {"
-                    + " active { name doDelete { name } } archived { data { name doDelete { name }"
-                    + " } aggregate { count } } } } }");
+                        + " active { data { name doDelete { name }"
+                        + " } aggregate { count } }"
+                        + " archived { data { name doDelete { name }"
+                        + " } aggregate { count } }"
+                        + " } } }");
```
Comment (andrewazores) [inline src/test/java/itest/GraphQLIT.java:468]: Do the sample queries in https://github.com/cryostatio/cryostat/tree/main/docs/graphql also need updating similarly to this change?

**Comment by andrewazores** [comment]
> (for me who doesn't know how to validate the SSL for 9095).

https://github.com/cryostatio/cryostatio.github.io/pull/70

+

https://github.com/andrewazores/vertx-fib-demo/blob/master/src/main/extras/app/resources/vertx-fib-demo.cer

File: src/main/java/io/cryostat/net/web/http/api/v2/graph/ActiveRecordingsFetcher.java (hunk 1 above)
Comment (andrewazores) [inline src/main/java/io/cryostat/net/web/http/api/v2/graph/ActiveRecordingsFetcher.java:165]:
```suggestion
        long count;
```

**Review by andrewazores — APPROVED** [review]

**Review by andrewazores — APPROVED** [review]

**Pull request merged**